packages/vaadin-themable-mixin/vaadin-themable-mixin.js

```diff
@@ -195,35 +199,34 @@ export const ThemableMixin = (superClass) =>
       super.finalize();
 
       const template = this.prototype._template;
-      if (!template || template.__themes) {
+      if (!template || hasThemes(this.is)) {
         return;
       }
 
-      const inheritedTemplate = Object.getPrototypeOf(this.prototype)._template;
-      const inheritedThemes = (inheritedTemplate ? inheritedTemplate.__themes : []) || [];
-
-      template.__themes = [...inheritedThemes, ...getThemes(this.is)];
-
-      // Get flattened styles array
-      const styles = template.__themes.reduce((styles, theme) => [...styles, ...theme.styles], []);
-      addStylesToTemplate(styles, template);
+      addStylesToTemplate(this.getStylesForThis(), template);
     }
 
     /**
      * Covers LitElement based component styling
      *
-     * NOTE: This is not yet an offically supported API!
-     *
-     * TODO: Add tests (run a variation of themable-mixin.test.js where the components get created as LitElements)
      * @protected
      */
     static finalizeStyles(styles) {
-      return (
-        getThemes(this.is)
-          // Get flattened styles array
-          .reduce((styles, theme) => [...styles, ...theme.styles], [])
-          .concat(styles)
-      );
+      return [styles, ...this.getStylesForThis()];
```

Comment:
Let's add a comment to clarify that this order means that styles defined in `static get styles()` tag of a component - those that it might define in `src` folder - come first, so that applied themes might override them.

Reply:
Done.